src/big/big_int.cr

```diff
@@ -267,8 +267,22 @@ struct BigInt < Int
     to_s io
   end
 
+  private def hash_normalize
+    if self == to_i64
```

Comment:
Comment to `Object#hash`:
> This method must have the property that `a == b` implies `a.hash == b.hash`

```
$ bin/crystal eval 'require "big"; b=BigInt.new(-1000); p b.to_i64, b.to_u64'
Using compiled compiler at `.build/crystal'
-1000
1000
```

So I can't use just `to_u64`. Nor I can just use `to_i64` cause of numbers between INT64MAX and UINT64_MAX. Although, probably I can use `self & 0xffffffffffffffff_u64`, but it will be tweaky.